blockchain/tx_pool.go

```diff
@@ -318,10 +318,14 @@ func (pool *TxPool) loop() {
 					continue
 				}
 				// Any non-locals old enough should be removed
-				if time.Since(pool.beats[addr]) > pool.config.Lifetime {
-					for _, tx := range pool.queue[addr].Flatten() {
-						pool.removeTx(tx.Hash(), true)
+				if beat, exist := pool.beats[addr]; exist {
+					if time.Since(beat) > pool.config.Lifetime {
+						for _, tx := range pool.queue[addr].Flatten() {
+							pool.removeTx(tx.Hash(), true)
+						}
 					}
+				} else {
+					pool.beats[addr] = time.Now()
```

Comment:
How about reducing the depths like below?

```
// Any non-locals old enough should be removed
beat, exist := pool.beats[addr]
if !exist {
  pool.beats[addr] = time.Now()
  continue
}

if time.Since(beat) > pool.config.Lifetime {
  for _, tx := range pool.queue[addr].Flatten() {
    pool.removeTx(tx.Hash(), true)
  }
}
```